### PR TITLE
feat(hydrated_cubit): hydrated storage compaction

### PR DESCRIPTION
**Pull request by orsenkucher**
## Description

With intention to address https://github.com/felangel/hydrated_bloc/issues/57 issue,  
add `compact` method to `HydratedStorage` class, but not `Storage` interface.  
So you'll need to downcast.

In addition to `compact`, introduce `compactionStrategy` for `HydratedStorage.build` function.
## Type of Change

- [x] ✨ New feature (non-breaking change which adds functionality)
- [ ] 🛠️ Bug fix (non-breaking change which fixes an issue)
- [ ] ❌ Breaking change (fix or feature that would cause existing functionality to change)
- [ ] 🧹 Code refactor
- [ ] ✅ Build configuration change
- [ ] 📝 Documentation
- [ ] 🗑️ Chore

### CODE REVIEW

**Comment by freetoken-Tg1483** [comment]
Build configuation change like everyone says hello im new making my first comments today.

**Comment by freetoken-Tg1483** [comment]
Non breaking chang🥰🥰

**Comment by freetoken-Tg1483** [comment]
Build a none breaking  configuation change  that has functionality existing features you can merg your branches

**Comment by orsenkucher** [comment]
@freetoken-Tg1483 I have found new friend for you🥳! His name is @bors

**Comment by orsenkucher** [comment]
Don't rly like what happened. Hive specifies that default `defaultCompactionStrategy` inside interface, and not doing `??= defaultCompactionStrategy` inside `HiveImpl`

File: packages/hydrated_cubit/test/hydrated_storage_test.dart [hunk 1]
```diff
@@ -68,6 +69,26 @@ void main() {
         expect(box, isNotNull);
         expect(box.path, p.join(directory.path, 'hydrated_box.hive'));
       });
+
+      test('usues defaultCompactionStrategy when needed', () async {
```
Comment (orsenkucher) [inline packages/hydrated_cubit/test/hydrated_storage_test.dart]: typo here

File: packages/hydrated_cubit/test/hydrated_storage_test.dart [hunk 2]
```diff
@@ -68,6 +69,26 @@ void main() {
...
+        storage = await HydratedStorage.build(compactionStrategy: null);
+        // TODO
+      });
+
+      // performCompactionIfNeeded
```
Comment (orsenkucher) [inline packages/hydrated_cubit/test/hydrated_storage_test.dart]: rm comment

File: packages/hydrated_cubit/test/hydrated_storage_test.dart [hunk 3]
```diff
@@ -68,6 +69,25 @@ void main() {
         expect(box, isNotNull);
         expect(box.path, p.join(directory.path, 'hydrated_box.hive'));
       });
+
+      test('uses defaultCompactionStrategy when needed', () async {
+        storage = await HydratedStorage.build(compactionStrategy: null);
+        // TODO
```
Comment (orsenkucher) [inline packages/hydrated_cubit/test/hydrated_storage_test.dart:75]: Have unfinished test here

File: packages/hydrated_cubit/test/hydrated_storage_test.dart [hunk 4]
```diff
@@ -135,6 +155,20 @@ void main() {
           verify(box.deleteFromDisk()).called(1);
         });
       });
+
+      group('compact', () {
+        test('not closed box', () async {
+          when(box.isOpen).thenReturn(false);
+          await (storage as HydratedStorage).compact();
+          verifyNever(await box.compact());
```
Comment (orsenkucher) [inline packages/hydrated_cubit/test/hydrated_storage_test.dart:163]: or throw an exception?

Comment (orsenkucher) [reply packages/hydrated_cubit/test/hydrated_storage_test.dart:163]: Nobody wants runtime exceptions, right?

**Comment by felangel** [comment]
@orsenkucher closing this since the repository has been moved to https://github.com/felangel/bloc. Let me know if you want me to migrate this PR or if you can. Thanks 🙏

**Pull request closed without merging**